modules/analysis-common/src/main/java/org/opensearch/analysis/common/MappingCharFilterFactory.java

```diff
@@ -77,10 +77,10 @@ public Reader create(Reader tokenStream) {
     private void parseRules(List<String> rules, NormalizeCharMap.Builder map) {
         for (String rule : rules) {
             Matcher m = rulePattern.matcher(rule);
-            if (!m.find()) throw new RuntimeException("Invalid Mapping Rule : [" + rule + "]");
```

Comment:
@nknize @adnapibar just a question: would sharing the path instead help? (rules are detach from source at this time but it is feasible to preserve this link). I am thinking if sharing that information would help in troubleshooting.

Reply:
+1 , and do you think logging at debug level would help the operator debug if needed?

Reply:
The path is not always available as the mapping rules can be included in the request. Also, the file path is passed in the request, so the user already knows the path.

Reply:
>  Also, the file path is passed in the request, so the user already knows the path.

You mean in the mappings? But there could be many of those, right? So which ones are problematic is not clear

Reply:
Yes, there can be many mapping rules. There are two ways you can supply those, 1) literally in the request 2) a path to the file in local file system.  
Ref: https://www.elastic.co/guide/en/elasticsearch/reference/7.10/analysis-stemmer-override-tokenfilter.html

Reply:
Thanks @adnapibar , I am fine with the change (it addresses the real issue), but we might leave users in the dark regarding where to look to fix the problem .... So up to you :-)